### PR TITLE
VZ-6158 - Update Keycloak to support multiple replicas and add affinity rules

### PR DESCRIPTION
**Pull request by mgianatagh**
Update Keycloak to support multiple replicas and add affinity rules

### CODE REVIEW

File: tests/e2e/pkg/kubernetes.go [hunk 1]
```diff
@@ -187,6 +203,21 @@ func GetDeployment(namespace string, deploymentName string) (*appsv1.Deployment,
 	return deployment, nil
 }
 
+// DoesStatefulSetExist returns whether a StatefulSet with the given name and namespace exists for the cluster
+func DoesStatefulSetExist(namespace string, name string) (bool, error) {
+	statefulsets, err := ListStatefulSets(namespace)
+	if err != nil {
+		Log(Error, fmt.Sprintf("Failed listing StatefulSets in cluster for namespace %s: %v", namespace, err))
```
Comment (kminder) [inline tests/e2e/pkg/kubernetes.go:210]: Possible a little too verbose (i.e. "in cluster" is unnecessary) but not a merge stopper.

Comment (stevshap) [inline tests/e2e/pkg/kubernetes.go:210]: "Failed to list" instead of "Failed listing" for consistency?

**Review by stevshap — DISMISSED** [review]

File: platform-operator/controllers/verrazzano/component/spi/testdata/basicDevMerged.yaml [hunk 2]
```diff
@@ -154,6 +154,19 @@ spec:
                       weight: 100
     keycloak:
       enabled: true
+      overrides:
+        - values:
+            replicas: 1
+            affinity: |
```
[the remark this answers is not in the document]
Comment (mgianatagh) [reply platform-operator/controllers/verrazzano/component/spi/testdata/basicDevMerged.yaml:177]: this is to match what gets generated from our merge logic.

File: tests/e2e/verify-install/verrazzano/verrazzano_test.go [hunk 3]
```diff
@@ -466,15 +467,17 @@ var _ = t.Describe("In Verrazzano", Label("f:platform-lcm.install"), func() {
 
 		t.It("has correct number of pods running", func() {
 			if isMinVersion110 {
-				validateCorrectNumberOfPodsRunning("istio-ingressgateway", constants.IstioSystemNamespace)
+				err := validateCorrectNumberOfPodsRunning("istio-ingressgateway", constants.IstioSystemNamespace)
```
[the remark this answers is not in the document]
Comment (mgianatagh) [reply tests/e2e/verify-install/verrazzano/verrazzano_test.go:507]: I guess it is not, the IDE was warning that there are unchecked error status.  But it turns out the function always returns nil.  So I will remove "error" from being returned.

File: tests/e2e/verify-install/verrazzano/verrazzano_test.go [hunk 4]
```diff
@@ -546,6 +548,39 @@ var _ = t.Describe("In Verrazzano", Label("f:platform-lcm.install"), func() {
...
+					err := validateCorrectNumberOfPodsRunningSts(constants.Keycloak, constants.KeycloakNamespace, "app.kubernetes.io/name")
+					Expect(err).To(BeNil())
+				} else {
+					t.Logs.Info("Skipping check, Verrazzano minimum version is not V1.1.0")
+				}
+			})
+
+			t.It("has affinity configured as expected", func() {
+				if isMinVersion140 {
+					assertPodAntiAffinity(map[string]string{"app.kubernetes.io/name": constants.Keycloak}, constants.KeycloakNamespace)
```
Comment (mbenmclean) [inline tests/e2e/verify-install/verrazzano/verrazzano_test.go:574]: This should only have affinity configured in the prod profile right? What happens when this runs against dev?

Comment (mgianatagh) [reply tests/e2e/verify-install/verrazzano/verrazzano_test.go:574]: affinity is set for all profiles

**Review by stevshap — APPROVED** [review]

**Pull request merged**